xwiki-platform-core/xwiki-platform-refactoring/xwiki-platform-refactoring-default/src/main/java/org/xwiki/refactoring/internal/ResourceReferenceRenamer.java

```diff
@@ -115,9 +146,18 @@ private boolean updateAbsoluteResourceReference(ResourceReference resourceRefere
             this.defaultReferenceDocumentReferenceResolver.resolve(linkEntityReference);
         EntityReference newTargetReference = newReference;
         ResourceType newResourceType = resourceReference.getType();
+        EntityReference absoluteResolvedReference = this.entityReferenceResolver.resolve(resourceReference, null);
 
-        // If the link targets the old (renamed) document reference, we must update it.
-        if (linkTargetDocumentReference.equals(oldReference)) {
+        // If the link targets the old (renamed) document reference and it's an absolute reference
+        // (i.e. its resolution without any given parameter gives same result than its resolution with the
+        // currentDocument) then we must update it
+        // We also update the link if it's not an absolute link but the current document is not part of the move job,
+        // as in this case there won't be any other call to perform the link refactoring.
+        boolean shouldBeUpdated = linkTargetDocumentReference.equals(oldReference)
+            && (absoluteResolvedReference.equals(linkEntityReference)
```

Comment:
So I did tested your scenario and the link is properly renamed, the condition here is respected. 
When I debug I have those values:
  * `resourceReference` is a typed `doc` reference of value `Source.Subspace.WebHome`
  * `currentDocumentReference` is a reference to `subwiki:Source.WebHome` 
  * `linkTargetDocumentReference` is resolved to `subwiki:Source.Subspace.WebHome`
  * `oldReference` is resolved to `subwiki:Source.Subspace.WebHome`
  * `absoluteResolvedReference` is resolved to `subwiki:Source.Subspace.WebHome`
  * `linkEntityReference` is resolved to `subwiki:Source.Subspace.WebHome`

> From what I understand, this means that links that are relative to the wiki won't be updated.

it works because the link is resolved using the current document so the proper subwiki is used.

Reply:
In the answers below to a question about what seems to be the exactly same "is absolute" condition in the other method, you write:

> the condition will never be met if I have a link to `xwiki:MySpace.MyPage`, but it will be met if I have `MySpace.MyPage`

However, here you argue that the condition is met for a link `Source.Subspace.WebHome`. So either I missed something or one of these answers is wrong.

Reply:
Still drinking my coffee so maybe it's me who is missing something, but the condition I was talking about in the other answer was actually the contrary of this one, there's a not: `!absoluteResolvedReference.equals(linkEntityReference)`.

Reply:
Yes, there is a negation, I was talking about the actual check for absolute references without negation.